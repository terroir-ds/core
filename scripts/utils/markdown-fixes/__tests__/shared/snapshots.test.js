import { describe, it, expect } from 'vitest';

// This would test against known good outputs
describe('markdown fixes snapshots', () => {
  describe('fixture tests', () => {
    it('should match snapshot for README formatting', () => {
      const input = `
# Project Title

## Installation
\`\`\`
npm install package
\`\`\`

## Usage
Basic usage:
\`\`\`
const pkg = require('package');
pkg.doSomething();
\`\`\`
## API

### method1()
\`\`\`
pkg.method1(options)
\`\`\`
Returns a promise.

### method2()
\`\`\`
pkg.method2(data)
\`\`\`
Processes data.
`;

      // After processing, this should produce a specific output
      // that we can snapshot test
      expect(input).toMatchSnapshot('readme-before');
      
      // const processed = processMarkdown(input);
      // expect(processed).toMatchSnapshot('readme-after');
    });

    it('should match snapshot for documentation formatting', () => {
      const input = `
# API Reference

## Functions

### \`processData(input)\`

Processes the input data.

**Parameters:**
- \`input\` - The data to process

**Example:**
\`\`\`
const result = processData({
  name: 'test',
  value: 42
});
\`\`\`

**Returns:**
\`\`\`
{
  processed: true,
  result: { ... }
}
\`\`\`

### \`validateInput(data)\`

Validates input data.

\`\`\`
if (!validateInput(data)) {
  throw new Error('Invalid input');
}
\`\`\`
`;

      expect(input).toMatchSnapshot('api-docs-before');
      
      // const processed = processMarkdown(input);
      // expect(processed).toMatchSnapshot('api-docs-after');
    });
  });
});