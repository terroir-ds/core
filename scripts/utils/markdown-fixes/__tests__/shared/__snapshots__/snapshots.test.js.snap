// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`markdown fixes snapshots > fixture tests > should match snapshot for README formatting > readme-before 1`] = `
"
# Project Title

## Installation
\`\`\`
npm install package
\`\`\`

## Usage
Basic usage:
\`\`\`
const pkg = require('package');
pkg.doSomething();
\`\`\`
## API

### method1()
\`\`\`
pkg.method1(options)
\`\`\`
Returns a promise.

### method2()
\`\`\`
pkg.method2(data)
\`\`\`
Processes data.
"
`;

exports[`markdown fixes snapshots > fixture tests > should match snapshot for documentation formatting > api-docs-before 1`] = `
"
# API Reference

## Functions

### \`processData(input)\`

Processes the input data.

**Parameters:**
- \`input\` - The data to process

**Example:**
\`\`\`
const result = processData({
  name: 'test',
  value: 42
});
\`\`\`

**Returns:**
\`\`\`
{
  processed: true,
  result: { ... }
}
\`\`\`

### \`validateInput(data)\`

Validates input data.

\`\`\`
if (!validateInput(data)) {
  throw new Error('Invalid input');
}
\`\`\`
"
`;
